minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/CsvMaker.java

```
@@ -0,0 +1,112 @@
+                .substance()
+                .flatMap(DatamartAllergyIntolerance.Substance::coding)
+                .flatMap(DatamartCoding::system)
+                .orElse(""));
+        csvOutput.add(
+            datamartAllergyIntolerance
+                .substance()
+                .flatMap(DatamartAllergyIntolerance.Substance::coding)
+                .flatMap(DatamartCoding::code)
+                .orElse(""));
+        csvOutput.add(
+            datamartAllergyIntolerance
+                .substance()
+                .flatMap(DatamartAllergyIntolerance.Substance::coding)
+                .flatMap(DatamartCoding::display)
+                .orElse(""));
+        csvOutput.add(datamartAllergyIntolerance.clinicalStatus().name());
+        csvOutput.add("");
+        csvOutput.add(
+            Objects.requireNonNull(datamartAllergyIntolerance.recordedDate().orElse(null))
+                .toString());
+        return csvOutput;
+      };
+
+  String resourceToWrite;
+
+  File dmDirectory;
+
+  CSVPrinter csvPrinter;
+
+  public CsvMaker(File dmDirectory, String resourceToWrite, CSVPrinter csvPrinter) {
+    this.dmDirectory = dmDirectory;
+    this.resourceToWrite = resourceToWrite;
+    this.csvPrinter = csvPrinter;
+    loadPatients(dmDirectory);
+  }
+
+  private <DM extends HasReplaceableId, E extends List<String>> void csvByResource(
+      File datamartDirectory, Class<DM> resourceType, Function<DM, E> toDatamartCsv) {
+    MakerUtils.findUniqueFiles(datamartDirectory, DatamartFilenamePatterns.get().json(resourceType))
+        .parallel()
+        .map(
+            f -> {
+              DM dm = MakerUtils.fileToDatamart(f, resourceType);
+              return toDatamartCsv.apply(dm);
+            })
+        .forEachOrdered(
+            s -> {
+              try {
+                csvPrinter.printRecord(s);
+              } catch (IOException e) {
+                log.error("Unable to output line {} to csv file!", s);
```

This block should rethrow the exception (as an unchecked exception), not swallow it. If something does go wrong, it should abort the whole process

However, this forced catch (since `@SneakyThrows` does not apply inside lambdas) is lame in general... how about collecting the `map()` results to a list and then doing `printRecord()` in a for loop instead, which will honor `@SneakyThrows`?